docs/guides/cloud-secrets.md

```diff
@@ -162,6 +177,16 @@ EARTHLY_TOKEN=...
 Which will then force Earthly to use that token when accessing secrets. This is useful for cases
 where running an ssh-agent is impractical.
 
+# Security Details
+
+The Earthly command uses HTTPS to communicate with the cloud secrets server. The server encrypts all
+secrets using OpenPGP's implemtation of AES256 before storing it in a database.
```

Comment:
Perhaps we should also mention that a distinct encryption key is used for each Earthly organization and that we use industry-standard security practices for managing encryption keys in the cloud.

Reply:
I took another look at our server code, and we don't have distinct per-organization encryption keys, but only a global key.

(as an aside, things like account passwords are hashed via bcrypt and can't be decrypted).

I'll add on `industry-standard security practices for managing encryption keys in the cloud` though.

Reply:
We should adjust this on the server side - mind adding an issue so we don't forget? The individual org encryption keys should be stored encrypted with the master key.

Reply:
Not something for this PR, but I think it's definitely worth considering support for client side encryption. This is beneficial in a few ways, such as the client not having to trust the server, but also then you no longer have to manage toxic assets like credentials for other org's critical / key systems.

For example, I can forsee that many orgs would like to store a github token as a secret so that they can greatly simplify checking out repositories. However, if it's server side only encryption, the client must now explicitly trust that the server (and in turn you) to keep it safe. Anyone with that token can access any of that orgs repos...

One way to do this without huge pain (even as a v1), would be to accept a symmetric key on the client that encrypts the secret with that key before uploading it. The server side stuff won't need to change - it just sends, receives and stores as normal - and on the client side, if the key is set, use that key to decrypt all the secrets that are coming in. Potentially could add a simple tag so the secret so that the client can identify explicitly that it's encrypted, and which key it needs to decrypt it. This also allows for mixed keys and mixing encrypted and non encrypted secrets.

Just some thoughts :)

Reply:
We went back and forth figuring out a way to do this and ultimately ended up with a system that does grant the server the ability to decrypt secrets in order to keep setting up CI tasks.

I don't see any technical issues with allowing users to specify a symmetric key for use strictly on the client side before calling the appropriate [set](https://github.com/earthly/earthly/blob/e8b0e570c5d843afad953ab8caab118d34adc229/secretsclient/client.go#L568) and [get](https://github.com/earthly/earthly/blob/e8b0e570c5d843afad953ab8caab118d34adc229/secretsclient/client.go#L561) functions. But the downside is now the symmetric key must be passed around different members of the team in addition to managing access to the secrets org. Then if you want to grant github actions access to secrets, you'll have to trust it and grant it access via an `EARTHLY_TOKEN` as well as this symmetric key.